kernal/x16/65c816/interrupt.s

```diff
@@ -173,15 +180,9 @@ __interrupt_65c816_native_kernal_impl_ret:
 	rti
 
 nnnmi:
-	.A16
-	.I16
-	sec
-	xce
-	.A8
-	.I8
-	clc
-	stz rom_bank
-	jmp (nminv)
+	c816_interrupt_impl
+	intr_common_impl
+	jmp nmi
```

Comment:
`nnnmi` needs to setup an emulation mode context and then jump to the `nminv` vector, not the KERNAL implementation; otherwise, NMIs in native mode won't ever execute a NMI handler written in emulation mode for the 65C02.

Reply:
This is exactly what is done here.  `nmi` is the target of the emulation mode NMI vector, and this is what does `jmp (nminv)`

Reply:
the code for the emulation mode nmi target is in `__nmi` in the source, as it is copied into low RAM at init.  The preservation that `nmi` does is minimal, so a double `pla` from a custom emulation mode NMI handler to clear the stack that it does should be able to `rti` into our fake RTI frames, one to return to native mode, and the other to return to the interrupted code.